.github/workflows/push-mimir-build-image.yml

```diff
@@ -0,0 +1,103 @@
+name: Build and Push mimir-build-image
+
+# configure trigger by pull request or manually
+on:
+  pull_request:
+    types: [opened, synchronize, ready_for_review]
+    paths:
+      - mimir-build-image/Dockerfile  
+
+jobs: 
+  build_and_push:
+    runs-on: ubuntu-latest
+    # If the PR is raised by dependabot, then build the image
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Checkout Pull Request Branch
+        run: gh pr checkout ${{ github.event.pull_request.number }}
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Prepare Variables
+        id: prepare
+        run: |
+          echo "path=mimir-build-image/Dockerfile" >> $GITHUB_OUTPUT
+          echo "image=grafana/mimir-build-image" >> $GITHUB_OUTPUT
+
+      - name: Compute Image Tag
+        id: compute_hash
+        run: |
+          current_hash=$(md5sum ${{ steps.prepare.outputs.path }} | awk '{print $1}')
+          echo "the file path is ${{ steps.prepare.outputs.path }}"
+          echo "build tag is $current_hash"
+          tag="${{ github.event.pull_request.number }}_$current_hash"
+          echo "tag=$tag" >> $GITHUB_OUTPUT
+
+      - name: Check Should Build Image
+        id: check_build
+        run: |
+          echo "Checking if image should be built"
+          image="docker.io/${{ steps.prepare.outputs.image }}:${{ steps.compute_hash.outputs.tag }}"
```

Comment:
nit: `skopeo inspect --raw "docker://grafana/mimir:latest"` seems to work too (ie. without `docker.io/` prefix). I'd suggest to use that to avoid hardcoding `docker.io` if it's not required.